Add `profile_ranking_avg` and `reduction_targets_avg` columns at company level in emission and sector profile indictors repectively

kalashsinghal opened the pull request:
closes #161 

Dear @AnneSchoenauer @Tilmon @maurolepore,
This PR adds the `profile_ranking_avg` and `reduction_targets_avg` columns at company level in emission and sector profile indictors respectively. The average column is the mean of `profile_ranking` values after grouping the rows based on columns `companies_id` and `group_by`. You can look at a small example using this reprex:

```
library(readr)
library(dplyr)
devtools::load_all(".")
#> ℹ Loading tiltIndicatorAfter
options(width = 500)

local_options(readr.show_col_types = FALSE)
companies <- read_csv(toy_emissions_profile_any_companies()) |>
    filter(companies_id %in% c("nonphilosophical_llama"))
  co2 <- read_csv(toy_emissions_profile_products_ecoinvent())
  europages_companies <- read_csv(toy_europages_companies())
  ecoinvent_activities <- read_csv(toy_ecoinvent_activities())
  ecoinvent_europages <- read_csv(toy_ecoinvent_europages())
  isic_name <- read_csv(toy_isic_name())

  out <- profile_emissions(
    companies,
    co2,
    europages_companies,
    ecoinvent_activities,
    ecoinvent_europages,
    isic_name
  )

  product <- unnest_product(out) |>
    filter(benchmark == "all")
  company <- unnest_company(out) |>
    filter(benchmark == "all")

product
#> # A tibble: 2 × 7
#>   companies_id           grouped_by risk_category profile_ranking clustered                   activity_uuid_product_uuid           co2_footprint
#>   <chr>                  <chr>      <chr>                   <dbl> <chr>                       <chr>                                        <dbl>
#> 1 nonphilosophical_llama all        low                     0.32 surface finishing, galvanic 833caa78-30df-4374-900f-7f88ab44075b         0.693
#> 2 nonphilosophical_llama all        low                     0.34 surface engineering         833caa78-30df-4374-900f-7f88ab44075b         0.693

company
#> # A tibble: 3 × 5
#>   companies_id           grouped_by risk_category value profile_ranking_avg
#>   <chr>                  <chr>      <chr>         <dbl>               <dbl>
#> 1 nonphilosophical_llama all        high              0               0.33
#> 2 nonphilosophical_llama all        medium            0               0.33
#> 3 nonphilosophical_llama all        low               1               0.33

```

Please do not get misled by the `0.33` values in risk category high and medium. As you can see that the `low` risk category will have the average value of `0.33` from values `0.32` and `0.34`.

----

TODO

- [x] [Link related issue/PR]([url](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword)). 
- [x] Describe the goal of the PR. Avoid details that are clear in the diff.
- [x] Mark the PR as draft.
- [x] [Include a unit test](https://code-review.tidyverse.org/reviewer/aspects.html#sec-tests).
- [x] Review your own PR in "Files changed".
- [x] Ensure the PR branch is updated.
- [x] Ensure the checks pass.
- [x] Change the status from draft to ready.
- [x] Polish the PR title and description.
- [x] Assign a reviewer.

EXCEPTIONS

- [ ] Slide here any item that you intentionally choose to not do.

maurolepore reviewed (approved):
LGTM. 

Very clean PR. Thanks @kalashsinghal

kalashsinghal commented:
Hi @AnneSchoenauer, Please have a look at the reprex for sector profile reduction targets:

``` r
library(readr)
library(dplyr)
devtools::load_all(".")
#> ℹ Loading tiltIndicatorAfter
options(width = 500)

local_options(readr.show_col_types = FALSE)
companies <- read_csv(toy_sector_profile_companies()) |>
  filter(companies_id %in% c("nonphilosophical_llama"))
scenarios <- read_csv(toy_sector_profile_any_scenarios())
europages_companies <- read_csv(toy_europages_companies())
ecoinvent_activities <- read_csv(toy_ecoinvent_activities())
ecoinvent_europages <- read_csv(toy_ecoinvent_europages())
isic_name <- read_csv(toy_isic_name())

out <- profile_sector(
  companies,
  scenarios,
  europages_companies,
  ecoinvent_activities,
  ecoinvent_europages,
  isic_name
)

product <- unnest_product(out) |>
  filter(scenario == "1.5C RPS", year == 2030)
company <- unnest_company(out) |>
  filter(scenario == "1.5C RPS", year == 2030)

product
#> # A tibble: 2 × 26
#>   companies_id           company_name           country sector_profile reduction_targets scenario  year ep_product                  matched_activity_name                                    matched_reference_product                     unit  tilt_sector tilt_subsector multi_match matching_certainty matching_certainty_company_average company_city postcode address                        main_activity activity_uuid_produc…¹ isic_4digit sector_scenario subsector_scenario ei_geography isic_4digit_name
#>   <chr>                  <chr>                  <chr>   <chr>                      <dbl> <chr>    <dbl> <chr>                       <chr>                                                    <chr>                                         <chr> <chr>       <chr>          <lgl>       <chr>              <chr>                              <chr>        <chr>    <chr>                          <chr>         <chr>                  <chr>       <chr>           <chr>              <chr>        <chr>           
#> 1 nonphilosophical_llama nonphilosophical_llama germany low                         0.08 1.5C RPS  2030 surface engineering         market for deep drawing, steel, 10000 kN press, automode deep drawing, steel, 10000 kN press, automode kg    metals      other metals   FALSE       low                low                                solms        35606    industriegebiet oberbiel | 35… distributor   833caa78-30df-4374-90… '2591'      industry        other industry     tilt_world   Forging, pressi…
#> 2 nonphilosophical_llama nonphilosophical_llama germany low                         0.10 1.5C RPS  2030 surface finishing, galvanic market for deep drawing, steel, 10000 kN press, automode deep drawing, steel, 10000 kN press, automode kg    metals      other metals   FALSE       low                low                                solms        35606    industriegebiet oberbiel | 35… distributor   833caa78-30df-4374-90… '2591'      industry        other industry     tilt_world   Forging, pressi…
#> # ℹ abbreviated name: ¹​activity_uuid_product_uuid

company
#> # A tibble: 3 × 13
#>   companies_id           company_name           country sector_profile_share sector_profile scenario year  matching_certainty_company_average company_city postcode address                                main_activity reduction_targets_avg
#>   <chr>                  <chr>                  <chr>                  <dbl> <chr>          <chr>    <chr> <chr>                              <chr>        <chr>    <chr>                                  <chr>                         <dbl>
#> 1 nonphilosophical_llama nonphilosophical_llama germany                    0 high           1.5C RPS 2030  low                                solms        35606    industriegebiet oberbiel | 35606 solms distributor                    0.09
#> 2 nonphilosophical_llama nonphilosophical_llama germany                    0 medium         1.5C RPS 2030  low                                solms        35606    industriegebiet oberbiel | 35606 solms distributor                    0.09
#> 3 nonphilosophical_llama nonphilosophical_llama germany                    1 low            1.5C RPS 2030  low                                solms        35606    industriegebiet oberbiel | 35606 solms distributor                    0.09
```

<sup>Created on 2024-02-27 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

AnneSchoenauer commented:
Great! This looks good. Please note that your example however is not really realistic as the two products you had chosen have the same tilt subsector but have different sector reduction targets with the same scenario providers - this should never be the case. Don’t know if you already want to write a test for this? But overall the mathematical results look fine!

kalashsinghal commented:
@AnneSchoenauer Yes you are right! Great review! :D

I changed/faked the values of reduction targets manually for you to see the calculation clearly. Otherwise both values were same `0.09` and the average of two equal values is the same value `0.09`. That's why I changed them. I hope its not an issue. BTW I also checked the function with different values and its not an issue :)
